lib/install/management/configure.go

```diff
@@ -53,43 +55,45 @@ var (
 )
 
 // Configure will try to reconfigure vch appliance. If failed will try to roll back to original status.
-func (d *Dispatcher) Configure(vch *vm.VirtualMachine, conf *config.VirtualContainerHostConfigSpec, settings *data.InstallerData, isConfigureOp bool) (err error) {
+func (d *Dispatcher) Configure(conf *config.VirtualContainerHostConfigSpec, settings *data.InstallerData) (err error) {
 	defer trace.End(trace.Begin(conf.Name, d.op))
 
-	d.appliance = vch
-
-	// update the displayname to the actual folder name used
-	if d.vmPathName, err = d.appliance.FolderName(d.op); err != nil {
-		d.op.Errorf("Failed to get canonical name for appliance: %s", err)
+	// set the folerName for ISO uploads
+	if d.vmPathName, err = d.appliance.DatastoreFolderName(d.op); err != nil {
+		d.op.Errorf("Failed to get the datastore folder name for the appliance: %s", err)
 		return err
 	}
 
 	ds, err := d.session.Finder.Datastore(d.op, conf.ImageStores[0].Host)
 	if err != nil {
-		err = errors.Errorf("Failed to find image datastore %q", conf.ImageStores[0].Host)
+		err = errors.Errorf("Failed to find the image datastore %q", conf.ImageStores[0].Host)
 		return err
 	}
 	d.session.Datastore = ds
 	d.setDockerPort(conf, settings)
 
 	if len(settings.ImageFiles) > 0 {
 		// Need to update iso files
-		if err = d.uploadImages(settings.ImageFiles); err != nil {
-			return errors.Errorf("Uploading images failed with %s. Exiting...", err)
+		if err = d.uploadISOs(settings.ImageFiles); err != nil {
+			return errors.Errorf("Uploading appliance isos failed with %s. Exiting...", err)
 		}
 		conf.BootstrapImagePath = fmt.Sprintf("[%s] %s/%s", conf.ImageStores[0].Host, d.vmPathName, settings.BootstrapISO)
 	}
 
-	if err = d.updateResourceSettings(conf.Name, settings); err != nil {
-		err = errors.Errorf("Failed to reconfigure resources: %s", err)
-		return err
-	}
-
-	defer func() {
-		if err != nil {
-			d.rollbackResourceSettings(conf.Name, settings)
+	// Resource Pools not available in a DRS Disabled environment, so only attempt an update
+	// if DRS is Enabled and this is a configure action.
+	if d.session.DRSEnabled != nil && *d.session.DRSEnabled && d.Action == ConfigureAction {
```

Comment:
Should we be rolling back resource changes for a rollback action as well?

Reply:
We do -- if DRS is enabled and we have an error we will roll back...if DRS is disabled we won't attempt a ResourceConfig since it's not valid.

Reply:
Oh wait..I think I misunderstood your question...you mean the rollback sub command?...tricky little devil...

If it's really a rollback I'd say yes...but currently we don't store that anywhere so we'd need to add that...I assume the previous thought was they could `configure` if they didn't like the `pool` settings...

Reply:
Should we open an issue to track this? (Note to self: this will _also_ be a concern for #7797.)

Rolling back non-settings changes is... hard.